paddle/framework/tensor_impl.h

```diff
@@ -131,7 +131,7 @@ inline Tensor Tensor::Slice(const int& begin_idx, const int& end_idx) const {
   PADDLE_ENFORCE_LT(begin_idx, end_idx,
                     "Begin index must be less than end index.");
   PADDLE_ENFORCE_NE(dims_[0], 1, "Can not slice a tensor with dims_[0] = 1.");
-  size_t base = product(dims_) / dims_[0];
+  size_t base = numel_ / dims_[0];
```

Comment:
I agree that we add `Tensor::numel()` as it is such a commonly used feature.

I noticed that in the tensor implementation, we use `Tensor::numel_`, but in other places, we call `Tensor::numel()`. This implies that that is a cache -- the former, and the latter returns the former. It would be great if we don't expose such implementation details to code readers. It looks to me that we can just call `Tensor::numel()` everywhere.

Reply:
Done.